test/utils/initMatchers.ts

```diff
@@ -103,15 +103,15 @@ declare global {
        * @example expect(() => render()).toWarnDev('single message')
        * @example expect(() => render()).toWarnDev(['first warning', 'then the second'])
        */
-      toWarnDev(messages?: string | readonly string[]): void;
+      toWarnDev(messages?: string | readonly (string | boolean)[]): void;
```

Comment:
Why is this needed?

Reply:
The type is not complete and I just found out because the autocomplete tests I added is in typescript.

Here is the logic of the matcher: https://github.com/siriwatknp/material-ui/blob/joy/autocomplete/test/utils/initMatchers.ts#L534-L537

```js
...
  // Ignore skipped messages in e.g. `[condition && 'foo']`
    const remainingMessages = expectedMessages.filter((messageOrFalse) => {
      return messageOrFalse !== false;
    });
```

The matcher can receive `boolean`.